### PR TITLE
Remove python 2 specific code

### PR DESCRIPTION
**Pull request by stephenworsley**
As requested in #3457, Python 2 specific code has been removed.

### CODE REVIEW

File: lib/iris/tests/unit/analysis/maths/test_divide.py [hunk 1]
```diff
@@ -28,10 +28,7 @@ class TestBroadcasting(tests.IrisTest_nometa,
                        CubeArithmeticBroadcastingTestMixin):
     @property
     def data_op(self):
-        try:
-            return operator.div
-        except AttributeError:
-            return operator.truediv
+        return operator.truediv
```
Comment (bjlittle) [inline lib/iris/tests/unit/analysis/maths/test_divide.py]: @stephenworsley It seems that the whole point of the `data_op` property was convenience i.e., to do the right thing across Python2 and Python wrt `div` and `truediv`.

Now that we're Python3, I'm just questioning the need for the existence of the `data_op` property. 

Can't we just use `operator.truediv` or `/` directly inline for the tests? i.e.,

```python
    def test_unmasked_div_zero(self):
        # Ensure cube behaviour matches numpy operator behaviour for the
        # handling of arrays containing 0.
        dat_a = np.array([0., 0., 0., 0.])
        dat_b = np.array([2., 2., 2., 2.])

        cube_a = Cube(dat_a)
        cube_b = Cube(dat_b)

        com = dat_b / dat_a
        res = self.cube_func(cube_b, cube_a).data

        self.assertArrayEqual(com, res)
```
Also drop the `import operator` as well :wink:

File: lib/iris/tests/unit/analysis/maths/test_divide.py [hunk 2]
```diff
@@ -42,10 +39,7 @@ def cube_func(self):
 class TestMasking(tests.IrisTest_nometa, CubeArithmeticMaskingTestMixin):
     @property
     def data_op(self):
-        try:
-            return operator.div
-        except AttributeError:
-            return operator.truediv
+        return operator.truediv
```
Comment (bjlittle) [inline lib/iris/tests/unit/analysis/maths/test_divide.py]: As the [above comment](https://github.com/SciTools/iris/pull/3507/files#r342056420).

File: lib/iris/tests/unit/analysis/maths/test_divide.py (hunk 1 above)
Comment (stephenworsley) [reply lib/iris/tests/unit/analysis/maths/test_divide.py]: So it looks like what's going on here is that these classes are actually inheriting a bunch of tests from https://github.com/SciTools/iris/blob/master/lib/iris/tests/unit/analysis/maths/__init__.py

`data_op` is an abstract property in the parent class where it is used in other tests https://github.com/SciTools/iris/blob/66423f8104f25cbea71bc470b5769eb6da876265/lib/iris/tests/unit/analysis/maths/__init__.py#L28-L31 It seems like the current usage makes sense in this context.

Comment (bjlittle) [reply lib/iris/tests/unit/analysis/maths/test_divide.py]: @stephenworsley Cool. 

I guess what I'm pushing is that the whole `data_op` property as a concept is now defunct and should be removed. For me it's simply technical debt that should be purged.

Comment (stephenworsley) [reply lib/iris/tests/unit/analysis/maths/test_divide.py]: I don't think it is defunct, it allows code reuse for the tests in [iris.tests.unit.maths.\_\_init__](https://github.com/SciTools/iris/blob/master/lib/iris/tests/unit/analysis/maths/__init__.py) over multiple different operations, see also addition for example https://github.com/SciTools/iris/blob/master/lib/iris/tests/unit/analysis/maths/test_add.py

Comment (bjlittle) [reply lib/iris/tests/unit/analysis/maths/test_divide.py]: Okay, so it's generic and not specific, thanks for clarifying :+1:

**Comment by bjlittle** [comment]
@stephenworsley So is this good to go? Do you have more changes in-bound?

**Comment by stephenworsley** [comment]
@bjlittle Thats all I can find for the moment, If we find any others they can go in another pull request.

**Comment by bjlittle** [comment]
@stephenworsley Great detective work, thanks :smile:

**Pull request merged**